src/js.rs

```diff
@@ -978,6 +978,114 @@ extern "C" {
     pub fn values(object: &Object) -> Array;
 }
 
+// Reflect
+#[wasm_bindgen]
+extern "C" {
+    pub type Reflect;
+
+    /// The static Reflect.apply() method calls a target function with arguments as specified.
+    /// 
+    /// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Reflect/apply
+    #[wasm_bindgen(static_method_of = Reflect)]
+    pub fn apply(target: &Function, this_argument: &JsValue, arguments_list: &Array) -> JsValue;
+
+    /// The static Reflect.construct() method acts like the new operator, but as a function.
+    /// It is equivalent to calling new target(...args). It gives also the added option to 
+    /// specify a different prototype.
+    /// 
+    /// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Reflect/construct
+    #[wasm_bindgen(static_method_of = Reflect)]
+    pub fn construct(target: &Function, arguments_list: &Array) -> JsValue;
+    #[wasm_bindgen(static_method_of = Reflect, js_name = construct)]
+    pub fn construct_with_new_target(target: &Function, arguments_list: &Array, new_target: &Function) -> JsValue;
```

Comment:
This will probably not be as good if there are multiple optional properties. If this is how optional properties will be handled it will have to be documented properly.